### PR TITLE
Apply styling on a per module basis

### PR DESCRIPTION
**Pull request by cgardens**
## What
* It was driving me crazy that building a module would cause styling changes on totally unrelated modules. 
* This led to a bunch of wasted time where accidental styling changes were committed.
* It also meant that a user had to run `./gradlew format` before each commit to make sure the global style was applied. this is an awful command to run because it requires _every_ python project to install linting dependencies individually. The time this takes just grows monotonically with the number of connectors.

## How
* Push styling into sub module. Each module is now responsible for applying its own styling to itself. So you can just run `./gradlew :<sub-module-name>:spotlessApply` for java or `./gradlew :<sub-module-name>:spotlessApply:airbytePythonApply` for python (might still need a small tweak to the python stuff so that it runs spotless and the python linter.
* You can still run `./gradlew format` if you want and its behavior will be the same as before.

## Tradeoffs
* I haven't really found any downsides of this approach yet. I think there may be some overhead to running spotless per module but it doesn't seem to have changed build times adversely on my local machine in any noticeable way. Maybe I'm missing something.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -15,85 +15,6 @@ if (!env.containsKey('VERSION')) {
     throw new Exception('Version not specified in .env file...')
 }
 
-def createLicenseWith = { File license, String startComment, String endComment, String lineComment ->
```
Comment (cgardens) [inline build.gradle:18]: pretty much lift and shift the spotless code into subprojects...

**Review by michel-tricot — COMMENTED** [review]
Are you sure that this change doesn't make the overall build longer?

Looks good otherwise!

**Review by sherifnada — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -15,85 +15,6 @@ if (!env.containsKey('VERSION')) {
...
-    ]
-
-    if (System.getenv().containsKey("CORE_ONLY")) {
-        excludes.add("airbyte-integrations/connectors")
-    }
-
-    return fileTree(dir: rootDir, include: pattern, exclude: excludes.collect {"**/${it}"})
-}
-
-spotless {
```
[the remark this answers is not in the document]
Comment (cgardens) [reply build.gradle:64]: i think the project method is scoped by where these methods are defined.

**Comment by cgardens** [comment]
@michel-tricot on the first build there's a bit of a cost. after that it seems the same. trying to understand a bit better if there's anything that can be done to speed up the task that is slow on the first build `:spotlessInternalRegisterDependencies`

```
CORE_ONLY=1 ./gradlew build
master (first build): 1m 46s
master (subsequent builds): 1m 11s
with module spotless (first build): 3m 44s
with module spotless (subsequent builds): 1m 6s
```

**Review by jrhizor — APPROVED** [review]
looks good except for the unformatted files.

**Comment by cgardens** [comment]
Out of ideas here. Left an issue with the spotless crew to see if they can help with the one slow step: https://github.com/diffplug/spotless/issues/846. Going to see if we can make any progress there before meging.

**Comment by cgardens** [comment]
This is just too slow. The build times are too terrible to move forward with this.

**Pull request closed without merging**